karavan-app/src/main/webui/src/projects/ProjectsPage.tsx

```diff
@@ -80,7 +83,23 @@ export class ProjectsPage extends React.Component<Props, State> {
     }
 
     onProjectDelete = (project: Project) => {
-        this.setState({isDeleteModalOpen: true, projectToDelete: project})
+        KaravanApi.getProjectPipelineStatus(project.projectId, this.props.config.environment, (status?: PipelineStatus) => {
+            if(status === undefined){
+                this.setState({isDeleteModalOpen: true, projectToDelete: project})
+            }else if(status?.result ==="Running"){
+                this.setState({isPodRunningModalOpen: true, projectToDelete: project})
+            }else if(status?.result ==="Succeeded"){
+                KaravanApi.getProjectPodStatuses(project.projectId, this.props.config.environment, (statuses: PodStatus[]) => {
```

Comment:
We should also check Deployments

Reply:
I thought about it when implementing but didn't do it because I thought checking for Deployment will be a duplicate as I am already checking for all the running pods.  Are you thinking of a scenario where all the pods are terminated but deployment is still there?

Reply:
If it is scaled to zero. No posd but deployment

Reply:
Implemented the following checks:

* If the pipeline is Running or Started, ask the user to delete the Deployment first.
* If any other status of the Pipeline or undefined, Check the deployment status.
* If deployment is undefined, Show the user, the delete Modal dialog,
* If deployment is there, the ask user to delete the deployment first.